src/AspNet.Security.OAuth.AdobeIO/AdobeIOAuthenticationOptions.cs

```diff
@@ -0,0 +1,39 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using static AspNet.Security.OAuth.AdobeIO.AdobeIOAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.AdobeIO
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="AdobeIOAuthenticationHandler"/>.
+    /// </summary>
+    public class AdobeIOAuthenticationOptions : OAuthOptions
+    {
+        public AdobeIOAuthenticationOptions()
+        {
+            ClaimsIssuer = AdobeIOAuthenticationDefaults.Issuer;
+            CallbackPath = AdobeIOAuthenticationDefaults.CallbackPath;
+
+            AuthorizationEndpoint = AdobeIOAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = AdobeIOAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = AdobeIOAuthenticationDefaults.UserInformationEndpoint;
+            Scope.Clear();
+            Scope.Add("openid,AdobeID");
```

Comment:
Is this one claim with a comma in the name, or two scopes? If the latter, then you need to add each one individually.

Reply:
If I do two separate adds, I get an invalid scope error from them. I have no idea why.

Reply:
My guess would be that they use commas to delimit scopes instead of spaces, in which case you need to override the `FormatScope()` method in the handler like this:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/17b82420c9c3482e17a5a0d95ba4ab66fce9c42c/src/AspNet.Security.OAuth.Deezer/DeezerAuthenticationHandler.cs#L141-L143

Then the options code becomes:

```csharp
Scope.Add("openid");
Scope.Add("AdobeID");
```

Reply:
Done.